packages/labs/context/src/lib/controllers/context-provider.ts

```diff
@@ -83,19 +104,52 @@ export class ContextProvider<T extends Context<unknown, unknown>>
     // The check on composedPath (as opposed to ev.target) is to cover cases
     // where the consumer is in the shadowDom of the provider (in which case,
     // event.target === this.host because of event retargeting).
-    if (ev.context !== this.context || ev.composedPath()[0] === this.host) {
+    const consumerHost = ev.composedPath()[0] as Element;
+    if (ev.context !== this.context || consumerHost === this.host) {
       return;
     }
     ev.stopPropagation();
-    this.addCallback(ev.callback, ev.subscribe);
+    this.addCallback(ev.callback, ev.subscribe, consumerHost);
+  };
+
+  public onProviderRequest = (
+    ev: ContextProviderEvent<Context<unknown, unknown>>
+  ): void => {
+    // Ignore events when the context doesn't match.
+    // Also, in case an element is a consumer AND a provider
+    // of the same context, we want to avoid the element to self-register.
```

Comment:
Nit: `avoid the element to self-register` -> `avoid the element self-registering`

No action required.

Reply:
Removed the code in question